File: database/src/main/java/com/coopra/database/data_access_objects/TrackDao.java
package com.coopra.database.data_access_objects;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.coopra.database.entities.Track;

@Dao
public interface TrackDao {
    @Query("SELECT * FROM track_table ORDER BY activity_created_at DESC")
    DataSource.Factory<Integer, Track> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Track... tracks);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Track track);

    @Query("DELETE FROM track_table")
    void deleteAll();
}

File: database/src/main/java/com/coopra/database/data_access_objects/TrackDao.kt
package com.coopra.database.data_access_objects

import androidx.paging.DataSource
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.coopra.database.entities.Track

@Dao
interface TrackDao {
    @Query("SELECT * FROM track_table ORDER BY activity_created_at DESC")
    fun getAll(): DataSource.Factory<Int, Track>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg tracks: Track)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(track: Track)

    @Query("DELETE FROM track_table")
    fun deleteAll()
}